packages/clay-autocomplete/stories/index.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayAutocomplete from '../src';
+import ClayDropDown from '@clayui/drop-down';
+import fuzzy from 'fuzzy';
```

Comment:
Since this is only used in our `stories`, do you think we really need it at all? We can just use `string.match()`

Reply:
I'd rather keep `fuzzy`, it handles the match by character, I'd probably have to create some more logic to handle it.

Reply:
Oh so is this logic a clay requirement? I would figure that the matching logic of the autocomplete would be up to the user.

Reply:
It's something provided by the Lexicon, but this should be customized so I'm thinking of making it as a low-level component `<ClayAutocomplete.Item value={item.name} match={value} />` replacing `<ClayDropDown.Item>`. What do you think?

Reply:
Oh yeah thats a good idea. Im assuming the `match` prop would be a regex that makes the portion of the text bold?

Reply:
Not really 😁. It would be the value entered in the input that will be used to compare, we would use `fuzzy`. What do you think?

Reply:
Oh I see, we could do that. One thought I have is, what if a user doesn't want the bold matching? This makes me think I would rather have a call back for formatting, this would then allow people to do other sorts of highlighting, like uppercase.

```jsx
<ClayAutocomplete.Item value={item.name} formatter={(val) => val.toUpperCase()} />
```
But now that I type that out, I wonder why we even pass in props...
```jsx
<ClayAutocomplete.Item>{item.name.toUpperCase()}</ClayAutocomplete.Item>
```

Reply:
I'm not sure about this but it would be interesting to add the `formatter` option, but we would probably have to create our own internal logic to allow this or offer the same API as `fuzzy` but would not allow too many things to just add wrappers between the texts.

> But now that I type that out, I wonder why we even pass in props...

If following this example I do not know if it would be necessary to use `ClayAutocomplete.Item`, I would probably only use `ClayDropDown.Item`.

Reply:
I'd be fine with just using `ClayDropDown.Item`.

Reply:
Summarizing just to see if I understood what we're waking up here:

- Create `<ClayAutocomplete.Item value={item.name} match={value} />` for just that case in bold.
- If someone wants to create something custom, use `<ClayDropDown.Item />`

Am I right or missing something? 🙂

Reply:
Yeah that sounds good to me! And we can always change it in the future if needed too.